### PR TITLE
GEOS-8204 Move users/roles REST from geofence-server to rest-config

### PR DESCRIPTION
**Pull request by NielsCharlier**

### CODE REVIEW

**Comment by NielsCharlier** [comment]
Still need a core geoserver reviewer to approve...

**Comment by tbarsballe** [comment]
There are now merge conflicts blocking this from being merged, please fix when able.

File: doc/en/user/source/rest/api/userrole.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .. _rest_api_user_roles:
 
-Users/Groups and Roles Rest API
-===============================
+Users/Groups and Roles
+======================
```
Comment (jodygarnett) [inline doc/en/user/source/rest/api/userrole.rst]: REST API examples are in rst, the actual endpoint documentation is in swagger.

We are going through and reviewing swagger docs on geoserver-devel prior to the release; if you can commit to taking part in that process I would be happy to merge this in for the code freeze.

File: src/restconfig/src/main/java/org/geoserver/rest/security/RolesRestController.java [hunk 2]
```diff
@@ -2,110 +2,105 @@
  * This code is licensed under the GPL 2.0 license, available at the root
  * application directory.
  */
-package org.geoserver.geofence.rest;
+package org.geoserver.rest.security;
```
Comment (jodygarnett) [inline src/restconfig/src/main/java/org/geoserver/rest/security/RolesRestController.java]: thanks for adopting new package structure

File: src/restconfig/src/main/java/org/geoserver/rest/security/RolesRestController.java [hunk 3]
```diff
@@ -2,110 +2,105 @@
...
-
-@Controller
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@RestController(value = "rolesRestController")
+@RequestMapping(path = RestBaseController.ROOT_PATH + "/security/roles")
```
Comment (jodygarnett) [inline src/restconfig/src/main/java/org/geoserver/rest/security/RolesRestController.java]: Public class javadoc?

File: src/restconfig/src/main/java/org/geoserver/rest/security/UsersRestController.java [hunk 4]
```diff
@@ -2,152 +2,140 @@
...
-import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
 
-@Controller
+@RestController(value = "usergroupRestController")
+@RequestMapping(path = RestBaseController.ROOT_PATH + "/security/usergroup")
```
Comment (jodygarnett) [inline src/restconfig/src/main/java/org/geoserver/rest/security/UsersRestController.java]: Public class javadoc?

File: src/restconfig/src/test/java/org/geoserver/rest/security/UserRoleRestControllerTest.java [hunk 5]
```diff
@@ -2,32 +2,33 @@
...
+import org.geoserver.rest.security.xml.JaxbRoleList;
+import org.geoserver.rest.security.xml.JaxbUser;
+import org.geoserver.rest.security.xml.JaxbUserList;
 import org.geoserver.security.validation.PasswordPolicyException;
+import org.geoserver.test.GeoServerTestSupport;
+import org.junit.Before;
 import org.junit.Test;
 
-public class UserRoleRestControllerTest extends GeofenceBaseTest {
+public class UserRoleRestControllerTest extends GeoServerTestSupport {
```
Comment (jodygarnett) [inline src/restconfig/src/test/java/org/geoserver/rest/security/UserRoleRestControllerTest.java]: Public class javadoc? Wait I do not care so much for a test case...

File: doc/en/user/source/rest/api/userrole.rst (hunk 1 above)
Comment (NielsCharlier) [reply doc/en/user/source/rest/api/userrole.rst]: @jodygarnett sorry, I somehow missed this review. Thanks. Understanding I now have to wait until after the code freeze for merge. So I should move this documentation to swagger, or how can I help? Thanks.

Comment (jodygarnett) [reply doc/en/user/source/rest/api/userrole.rst]: Code is not frozen, will discuss in geoserver meeting.

As above likely we can merge if you can get the docs in before the release

**Comment by tbarsballe** [comment]
@jodygarnett - Is this fine to merge as-is (once conflicts are fixed), as long as we get swagger docs in before the release? Otherwise, it won't make it into the beta.

**Comment by jodygarnett** [comment]
Thanks @tbarsballe - as long as someone writes docs I am good

**Comment by NielsCharlier** [comment]
@jodygarnett is the deadliine for these docs this weekend?

**Comment by tbarsballe** [comment]
As long as the docs get done before the 2.12.0 release, it should be fine. That is currently scheduled for the 18th, but will almost certainly get pushed back a bit, given the delays getting the beta done.

**Comment by NielsCharlier** [comment]
Okay, if it doesn't get pushed back I will still do it this weekend.

**Pull request merged**